pkg/controlplane/instance.go

```diff
@@ -754,3 +787,30 @@ func DefaultAPIResourceConfigSource() *serverstorage.ResourceConfig {
 
 	return ret
 }
+
+// Utility function to check if the UnknownVersionInteroperabilityProxy feature flag is enabled.
+// Since UnknownVersionInteroperabilityProxy feature depends on APIServerIdentity and StorageVersionAPI
+// features, this function checks if all the three flags are enabled.
+func PeerProxyFeatureEnabled() bool {
+	if utilfeature.DefaultFeatureGate.Enabled(apiserverfeatures.UnknownVersionInteroperabilityProxy) {
+		if utilfeature.DefaultFeatureGate.Enabled(apiserverfeatures.APIServerIdentity) && utilfeature.DefaultFeatureGate.Enabled(apiserverfeatures.StorageVersionAPI) {
+			return true
+		} else {
+			klog.V(3).Infof("UnknownVersionInteroperabilityProxy feature requires both APIServerIdentity and StorageVersionAPI feature flag to be enabled")
+			return false
+		}
+	}
+	return false
+}
+
+// utility function to get the apiserver address that is used by peer apiservers to proxy
+// requests to this apiserver in case the peer is incapable of serving the request
+func getPeerAddress(peerBindAddress peerproxy.PeerBindAddress, publicAddress net.IP) string {
+	var ip string
```

Comment:
ip is misleading use `host` or `destination`or just not use a temporal variable and return directly the string

Reply:
done